tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp

```diff
@@ -452,6 +452,34 @@ class AstNamedCastExpr : public AstExpr {
   }
 };
 
+template <typename T>
+void DumpList(
+    T start,
+    T end,
+    AstDumper* dumper,
+    DumpNodeOptions const& dumpOptions,
+    std::function<void(AstDumper*, decltype(*start)&)> function)
```

Comment:
Is this a better name?
```suggestion
    std::function<void(AstDumper* dumper, decltype(*start)& element)> doDump(List)Element)
```

Reply:
Hmm. I didn't think you could have type names in the std::function template parameter. But it does seem to work.